graph/number_of_islands_dfs.cpp

```diff
@@ -0,0 +1,113 @@
+/**
+ * @file
+ *
+ * @brief Algorithm to count the number of islands
+ *
+ * @details
+ * Depth First Traversal (or Search) for a graph is similar to Depth First
+ * Traversal of a tree. The only catch here is, that, unlike trees, graphs may
+ * contain cycles (a node may be visited twice). To avoid processing a node more
+ * than once, use a boolean visited array. A graph can have more than one DFS
+ * traversal.
+ *
+ * The algorithm implemented in this file determines the number of islands in a
+ * 2D array of 1s(land) and 0s(water).
+ *
+ * @author [Harsh Agrawal](https://github.com/harshagrawal523)
+ *
+ */
+#include <iostream>  // for IO operations
+#include <string>    // for strings
+#include <vector>    // for vectors
+
+/**
+ * @namespace graph
+ * @brief Graphical algorithms
+ */
+namespace graph {
+/**
+ * @namespace number_of_islands
+ * @brief Functions for counting the number of islands
+ */
+namespace number_of_islands {
+/**
+ * @brief Class for representing the matrix of 0s(water) and 1s(land).
+ */
+class matrix {
+ private:
+    int n, m;  ///< dimensions of the graph
```

Comment:
Shouldn't we initialize this to `0`?
```suggestion
    int n = 0, m = 0;  ///< dimensions of the graph
```